### PR TITLE
Portmaster vs GlassWire blog post

### PR DESCRIPTION
**Pull request by davegson**

### CODE REVIEW

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+To detect trackers and bad domains, Portmaster uses filter lists. These lists are similar to lists used by tracker blockers for browsers like uBlock Origin. These lists contain domains discovered to track users and are auto-updated on a regular basis by open source communities around the world. Within Portmaster these filter lists can be customized to your needs by de/selecting different categories such as ads, trackers, malware, NSFW, etc.
+
+Just like GlassWire, Portmaster allows you to configure different settings for different network environments. This way you can easily switch from your trusted Home Wi-Fi to an untrusted network, like in a coffee shop.
+
+Portmaster also allows you to log and display different metrics about different apps. Like which domains, IP addresses they are connecting to and in which country the servers are located, along with the time taken to make the connections.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:152]: "time taken to make the connections" sounds like this is the time the connection needed to be established. While we maybe could get that information, we currently don't have it and don't show it.
We just show when a connection started and when it ended.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Having a business models and openly communicating this is an important thing for privacy tools. Because without a business model, a project might die or be bought out by Venture Capital, which have a tendency to be quick on monetizing intimate user data.
+
+### Portmaster and GlassWire Provide a Simple Dashboard for Detecting and Responding to Privacy Threats
+
+As far as network monitoring is concerned, the dashboard is an important part that will allow you to understand the network activity of your device. You would want to know which domains/websites were allowed to connect and which were blocked while you enjoyed your memes. In this case, Portmaster and GlassWire take a similar approach. They display all connections on a simple and easy to understand dashboard. GlassWire uses graphs, while Portmaster uses numbers. These connections are also sorted per app to help you differentiate connections made by your browser from the connections made by your chat app.
+
+When it comes to connection details, both Portmaster and GlassWire show the type of traffic, the IP address and country of the connected servers, the time taken to make a connection and similar metrics.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:170]: Same as before: "time taken to make a connection"

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+### Portmaster and GlassWire Allow You to Configure Settings System-Wide and Per-App
+
+Portmaster has global settings which define behavior for the whole device. But sometimes, an application may break due to a blocked connection. Allowing that connection system-wide might hurt your privacy. With Portmaster, you can easily solve this problem by creating an exception for a specific application, leaving other apps unaffected.
+
+You can even block risky connection types system-wide, such as p2p or incoming, and then create exceptions for trusted apps.
+
+### GlassWire Seems to Be A Proxy for the Windows Filtering Platform
+
+Disclaimer: GlassWire is closed-source - therefore we had to analyze their software in a blackbox test. We do not have the full picture on how GlassWire integrates into the operating system.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:182]: Would suggest:
We do not claim to have the full picture on how GlassWire integrates into the operating system.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+What we did find out is that GlassWire uses the Windows Filtering Platform (WFP) in user-space mode, creating rules for applications essentially exactly like the Windows Firewall does itself. This is a similar approach to that of [Simplewall]({{ site.simplewall_comparison_url }}). When Windows Firewall is disabled, this also makes GlassWire blind and incapable of blocking anything.
+
+GlassWire also uses a background service to collect additional data about connections. It is hard to find out what this service exactly does.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:186]: I would omit "It is hard to find out what this service exactly does." - we just stated that before.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+Finally, an oddity we came across was that GlassWire provides a tray icon with the "Exit" option. But pressing that does not end GlassWire or the service as expected - instead the software continues to do what it does. We feel that this is unfavorable wording and might not align with user expectation.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:188]: What I think it does that "Exit" stops the blocking of apps, but the collection of data continues. Sorry if that was not clear from the initial analysis.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+Contrary to GlassWire, Portmaster's system integration is both [documented](https://docs.safing.io/portmaster/architecture/os-integration#windows) and [open source](https://github.com/safing/portmaster-windows-kext/), so you can verify our claims.
+
+### Portmaster Safeguards Your DNS queries From ISPs to Enhance Your Privacy
+
+Encrypted DNS queries allow you to bypass Internet Service Providers (ISPs) and eavesdroppers and directly resolve your requests at the DNS provider. While the ISPs or middleware can no longer easily determine the websites you browse or the apps you use, but your DNS provider definitely can. That being said, even if your ISP is uninformed of the websites you visit, they can monitor the IP addresses you connect to. Since many services employ dedicated static IPs for their infrastructure, ISPs can try to track your queries using conditional logic. So even though DNS encryption improves your privacy, it cannot safeguard all your connections.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:200]: This just looks at the privacy aspect of encrypted DNS, but there are important security (and therefore, also privacy) benefits.
Also, 3/4 of this paragraph is disclaiming what secure DNS cannot do. This seems odd when reading.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+      </tr>
+      <tr class="card-table-body-row">
+        <td class="card-table-body-row-cell-title">Protection Type</td>
+        <td class="card-table-body-row-cell"><strong>Device</strong></td>
+        <td class="card-table-body-row-cell"><strong>Device</strong></td>
+      </tr>
+      <tr class="card-table-body-row">
+        <td class="card-table-body-row-cell-title">System Integration</td>
+        <td class="card-table-body-row-cell"><strong>Kernel Extension</strong></td>
+        <td class="card-table-body-row-cell"><strong>Windows Filtering Platform</strong></td>
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:54]: We also use WFP, but in the kernel, if you feel that is important.

File: _config.yml [hunk 8]
```diff
@@ -42,6 +42,7 @@ vpn_comparison_url: /blog/2020/01/22/how-the-spn-compares-to-vpns/
 tor_comparison_url: /blog/2020/01/22/how-the-spn-compares-to-tor/
 pihole_comparison_url: /blog/2021/12/09/portmaster-vs-pi-hole/
 simplewall_comparison_url: /blog/2022/04/11/portmaster-vs-simplewall/
+glasswire_comparison_url: /blog/2022/07/26/portmaster-vs-glasswire/
```
Comment (dhaavi) [inline _config.yml:45]: Change date.

File: _publications/2022-07-26-portmaster-vs-glasswire.md (hunk 2 above)
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:170]: Same as before: "time taken to make a connection"

File: _publications/2022-07-26-portmaster-vs-glasswire.md (hunk 3 above)
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:182]: Would suggest:
We do not claim to have the full picture on how GlassWire integrates into the operating system.

File: _publications/2022-07-26-portmaster-vs-glasswire.md (hunk 4 above)
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:186]: I would omit "It is hard to find out what this service exactly does." - we just stated that before.

File: _publications/2022-07-26-portmaster-vs-glasswire.md (hunk 5 above)
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:188]: What I think it does that "Exit" stops the blocking of apps, but the collection of data continues. Sorry if that was not clear from the initial analysis.

File: _publications/2022-07-26-portmaster-vs-glasswire.md (hunk 6 above)
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:200]: This just looks at the privacy aspect of encrypted DNS, but there are important security (and therefore, also privacy) benefits.
Also, 3/4 of this paragraph is disclaiming what secure DNS cannot do. This seems odd when reading.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+Contrary to GlassWire, Portmaster's system integration is both [documented](https://docs.safing.io/portmaster/architecture/os-integration#windows) and [open source](https://github.com/safing/portmaster-windows-kext/), so you can verify our claims.
+
+### Portmaster Safeguards Your DNS queries From ISPs to Enhance Your Privacy
+
+Encrypted DNS queries allow you to bypass Internet Service Providers (ISPs) and eavesdroppers and directly resolve your requests at the DNS provider. While the ISPs or middleware can no longer easily determine the websites you browse or the apps you use, but your DNS provider definitely can. That being said, even if your ISP is uninformed of the websites you visit, they can monitor the IP addresses you connect to. Since many services employ dedicated static IPs for their infrastructure, ISPs can try to track your queries using conditional logic. So even though DNS encryption improves your privacy, it cannot safeguard all your connections.
+
+We recommend using Tor, a trusted VPN or the SPN, which will provide additional privacy from your ISP.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:202]: I would emphasize that the SPN is a Portmaster Add-On or similar. I think this is important in this context.

File: _publications/2022-07-26-portmaster-vs-glasswire.md [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+
+Nevertheless, you get to enjoy enhanced security by preventing threats like DNS based Man-in-the-Middle (MitM) attacks and additional privacy from middleware. GlassWire needs extra software/setup to encrypt DNS queries, while Portmaster encrypts your DNS queries by default.
```
Comment (dhaavi) [inline _publications/2022-07-26-portmaster-vs-glasswire.md:204]: I would merge the first sentence into the first paragraph of this section in order to have benefits in one place.

**Review by dhaavi — CHANGES_REQUESTED** [review]
Looks good overall.
Some comments.
Only critical issues are the two "time taken to make a connection".

File: _publications/2022-07-26-portmaster-vs-glasswire.md (hunk 7 above)
Comment (davegson) [reply _publications/2022-07-26-portmaster-vs-glasswire.md:54]: I feel that is elaborated in the paragraphs

**Pull request merged**